docs/architecture/adr-0001-app-setup-and-goals.md

```diff
@@ -0,0 +1,168 @@
+contracts we need to enforce for the other kinds of models and is sufficient for
+object detection and chip classification tasks. `Model`s know what sorts of imagery
+are acceptable for them to obtain meaningful results:
+
+- an array of bandwidth requirements
+- a minimum spatial resolution
+- an area of interest
+
+We can determine this information for any incoming imagery that adheres to the
+`eo` spec using band information.
+
+`Model`s also need to know their own requirements to run. In particular, they
+need sufficient information to be able to fill in a `SubmitJob` request for `AWS`
+Batch. This includes resource requirements, a docker container accessible from
+within the hosting AWS account, a compute environment name, and a command to run.
+We require `model`s to carry this information to avoid having to infer it or
+guess at it based on heuristics.
+
+Once we identify that a `model` should run against an image, we'll kick off an
+AWS Batch job with the model's parameters. The `model`'s command _must_ accept a
+STAC item with the `eo` extension as its last parameter.
+
+`Model`s will be renamed to something less overloaded for developer sanity at
+some later date.
+
+### Notifying the API
+
+Since `model`s will be run in some asynchronous workflow, they'll need a way to
+notify the API. The form that notification should take is a [STAC item with the labeling extension](https://github.com/radiantearth/stac-spec/tree/master/extensions/label).
+Reasons to lean on STAC here are similar here to the reasons for leaning on it for new imagery. The API will accept a `POST` to a particular model
+that contains a STAC item with an asset that points to predictions on the new im
+agery.
+
+It's possible that future work could allow discrimination between predictions th
+at need time to run an asynchronous process and models that can do synchronous
+prediction. Since we don't know this is necessary or even possible, we'll assume
+for now that all predictions happen asynchronously and that all updates to the
+Granary API will happen via a later `POST` request.
+
+### Notifying end users
+
+When we create `model`s, we will also create AWS Simple Notification Service
+topics for those models. For our purposes we can think of these as an
+inexhaustible resource, since each AWS account is allowed 100,000 topics. At a
+time in the future when we think we are in danger of going over that limit we
+can think about a new strategy or ask for a service limit increase.
+Whenever a model is updated -- either a new prediction run has been kicked off,
+results are available, or metadata about that model has changed -- we'll send
+an event to the SNS topic for that model. The application will be deployable
+in any AWS account (including a client's account), so we assume that clients
+also have the ability to subscribe to those SNS topics and can subscribe
+to notifications.
+
+### Impact of continuous deployment
```

Comment:
This really isn't an actionable comment, but I wanted to surface that I am onboard with this.

My only request is that we all maintain a high level of awareness about the pros/cons of adopting this methodology _within this specific context_. CD is consistently put on a high pedestal in our industry (and for good reasons), but I am sensitive to how susceptible it is to cargo culting due to how vague its definition can be, and how its value can fluctuate across contexts.